### PR TITLE
chore: add python deprecation notice to help and init

### PR DESCRIPTION
**Pull request by nathan-weinberg**
we've seen a lot of confusion from folks using deprecated Python versions installing 0.17 - adding a message to recommend folks use a support Python version so they can use the latest ilab

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the
  [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [ ] [Changelog](https://github.com/instructlab/instructlab/blob/main/CHANGELOG.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Functional tests have been added, if necessary.
- [ ] E2E Workflow tests have been added, if necessary.

### CODE REVIEW

**Review by cdoern — APPROVED** [review]

File: src/instructlab/config/init.py [hunk 1]
```diff
@@ -70,6 +70,14 @@ def init(
             )
             if not overwrite:
                 return
+        click.secho(
+            """
+            NOTE: ilab no longer supports Python 3.9 or Python 3.12 as of version 0.18.
+            If are using either of these python versions, please switch to Python 3.10 or 3.11 to use the latest ilab.
```
Comment (kelbrown20) [inline src/instructlab/config/init.py:76]: Just a nit
```suggestion
            If you are using either of these Python versions, please switch to Python 3.10 or 3.11 to use the latest ilab.
```

File: src/instructlab/lab.py [hunk 2]
```diff
@@ -93,7 +93,12 @@ def format_epilog(self, ctx, formatter):
 @click.pass_context
 # pylint: disable=redefined-outer-name
 def ilab(ctx, config_file):
-    """CLI for interacting with InstructLab.
+    """
+    NOTE: ilab no longer supports Python 3.9 or Python 3.12 as of version 0.18.
+    If are using either of these python versions, please switch to Python 3.10 or 3.11 to use the latest ilab.
```
Comment (kelbrown20) [inline src/instructlab/lab.py:98]: Same here

File: src/instructlab/config/init.py [hunk 3]
```diff
@@ -70,6 +70,14 @@ def init(
...
             if not overwrite:
                 return
+        click.secho(
+            """
+            NOTE: ilab no longer supports Python 3.9 or Python 3.12 as of version 0.18.
+            If you are using either of these Python versions, please switch to Python 3.10 or 3.11 to use the latest ilab.
+            You can check which Python version you are using by running `python --version`
+            """,
+            fg="yellow",
+        )
```
Comment (mairin) [inline src/instructlab/config/init.py:80]: Suggest adding a line about consequences: "You can continue to run ilab on an unsupported version of Python, but you will be unable to upgrade to more recent versions of ilab with access to bug fixes and new features."

Comment (nathan-weinberg) [reply src/instructlab/config/init.py:80]: Updated the text, lmk what you think!

File: src/instructlab/config/init.py [hunk 4]
```diff
@@ -70,6 +70,14 @@ def init(
             )
             if not overwrite:
                 return
+        click.secho(
+            """
+            NOTE: ilab no longer supports Python 3.9 or Python 3.12 as of version 0.18. If you are using either of these Python versions, there is no upgrade path from ilab 0.17 to newer lab versions.
```
Comment (kelbrown20) [inline src/instructlab/config/init.py:75]:
```suggestion
            NOTE: ilab no longer supports Python 3.9 or Python 3.12 as of version 0.18. If you are using either of these Python versions, there is no upgrade path from ilab 0.17 to newer ilab versions.
```

**Review by kelbrown20 — APPROVED** [review]

**Pull request merged**